rllib/algorithms/algorithm.py

```diff
@@ -2701,7 +2707,11 @@ def _run_one_evaluation(
             if self.evaluation_workers is not None
             else 0
         )
-        eval_results["evaluation"]["num_recreated_workers"] = num_recreated
+        # Worker failures might have already been handled within `self._evaluate_async`
```

Comment:
This count would be overridden by 0 for `enable_async_evaluation=True`.

Reply:
🤔 wonder if we should move this try_recover_from_step_attempt() into self.evaluate() then.
or, I actually need to refactor all the remote gets anyways for elastic training. so maybe I can clean this up when I get to remote_req_manager.